src/runtime/pkg/kata-monitor/metrics.go

```diff
@@ -74,8 +74,14 @@ func registerMetrics() {
 
 // getMonitorAddress get metrics address for a sandbox, the abstract unix socket address is saved
 // in `metrics_address` with the same place of `address`.
-func (km *KataMonitor) getMonitorAddress(sandboxID, namespace string) (string, error) {
-	path := filepath.Join(km.containerdStatePath, types.ContainerdRuntimeTaskPath, namespace, sandboxID, "monitor_address")
+func (km *KataMonitor) getMonitorAddress(sandboxID, runtime string) (string, error) {
+	path := filepath.Join("/run/containerd", types.ContainerdRuntimeTaskPath, k8sContainerdNamespace, sandboxID, "monitor_address")
+	if runtime == RuntimeCRIO {
+		path = filepath.Join("/run/containers/storage/overlay-containers", sandboxID, "userdata", "monitor_address")
```

Comment:
I don't have enough background, but can we hope all the systems will follow the same path?